src/ORM/Queries/SQLSelect.php

```diff
@@ -250,14 +249,6 @@ public function setLimit($limit, $offset = 0)
             throw new InvalidArgumentException("SQLSelect::setLimit() only takes positive values");
         }
 
-        if ($limit === 0) {
-            Deprecation::notice(
```

Comment:
I've raised a new card for implementing this logic https://github.com/silverstripe/silverstripe-framework/issues/10604